src/mapeo-manager.js

```diff
@@ -199,6 +215,35 @@ export class MapeoManager extends TypedEmitter {
     return this.#replicate(noiseStream)
   }
 
+  /**
+   * @param {'blobs' | 'icons' | 'maps'} mediaType
+   * @returns {Promise<string>}
+   */
+  async #getMediaAddress(mediaType) {
+    /** @type {string | null} */
+    let prefix = null
+
+    switch (mediaType) {
+      case 'blobs': {
+        prefix = BLOBS_PREFIX
+        break
+      }
+      case 'icons': {
+        prefix = ICONS_PREFIX
+        break
+      }
+      default: {
+        throw new Error(`Unsupported media type ${mediaType}`)
+      }
+    }
+
+    const base = await pTimeout(getFastifyServerAddress(this.#fastify.server), {
+      milliseconds: 1000,
+    })
```

Comment:
if implementing the suggestion above. Also maybe 5 second timeout is safer (in case on a slow phone it takes a while for the server to start)

```suggestion
    const base = await getFastifyServerAddress(this.#fastify.server, {
      timeout: 5000,
    })
```

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/440/commits/3aafbc01327688488971d330e4d98482ac2d93b1